server/service/client_scripts.go

```diff
@@ -139,3 +143,78 @@ func (c *Client) ApplyNoTeamScripts(scripts []fleet.ScriptPayload, opts fleet.Ap
 
 	return resp.Scripts, err
 }
+
+func (c *Client) validateMacOSSetupScript(fileName string) ([]byte, error) {
+	if err := c.CheckAppleMDMEnabled(); err != nil {
+		return nil, err
+	}
+
+	b, err := os.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func (c *Client) deleteMacOSSetupScript(teamID *uint) error {
+	var query string
+	if teamID != nil {
+		query = fmt.Sprintf("team_id=%d", *teamID)
+	}
+
+	verb, path := "DELETE", "/api/latest/fleet/setup_experience/script"
+	var delResp deleteSetupExperienceScriptResponse
+	return c.authenticatedRequestWithQuery(nil, verb, path, &delResp, query)
+}
+
+func (c *Client) uploadMacOSSetupScript(filename string, data []byte, teamID *uint) error {
+	// there is no "replace setup experience script" endpoint, and none was
+	// planned, so to avoid delaying the feature I'm doing DELETE then SET, but
+	// that's not ideal (will always re-create the script when apply/gitops is
+	// run with the same yaml). Note though that we also redo software installers
+	// downloads on each run, so the churn of this one is minor in comparison.
```

Comment:
Wouldn't get existing script and compare be better?

Please file a bug for this. This will cause setup experience to malfunction in some rare cases, right?

Reply:
> Wouldn't get existing script and compare be better?

Not necessarily, we'd be getting from the reader and loading/comparing a potentially large string to check for differences. I agree that it's worth creating a ticket, as gitops is all about "replace whatever's there" so we don't need to pay the price of a check. I'll file this bug.

Reply:
> This will cause setup experience to malfunction in some rare cases

I don't think so? I mean, there's a slight race between the delete and the insert if something else re-inserts first, but then it would fail applying it. I don't think it can really malfunction because of this, but it's still not ideal IMO.